### DISCUSSION TITLE
Does LocalAI support models other than bert-MiniLM-L6-v2q4_0?

### DISCUSSION THREAD

**Discussion by zenyanbo** [mudler/LocalAI#1743] [Q&A] [answered]
Hello, Does LocalAI support models other than bert-MiniLM-L6-v2q4_0? 
For example, bge-base-en-v1.5, nomic-embed-text-v1.5. Because the dimensions and maximum token of all-MiniLM-L6-v2 are too small, only 384 and 512 respectively. Might not be enough for RAG.

**Comment by mudler** [comment]
yes, it does support anything supported by sentencetransformers and transformers for embeddings - you need to switch the model name appropriately in the YAML config file. See also the docs here: https://localai.io/features/embeddings/#huggingface-embeddings

**Reply by zenyanbo** [reply]
Thank you.  I'm always trying bert-embedding and llama.cpp embedding, sentencetransformers haven't tried yet.

**Comment by zenyanbo** [comment, marked as answer]
I found bge-m3 model which has 8k size and 1024 dimensions. I think it's useful.
Config:
```
name: text-embedding-3-large
backend: sentencetransformers
embeddings: true
parameters:
  model: BAAI/bge-m3
```

**Comment by ciekawy** [comment]
I managed to extend transformers backend and use bge-m3 as both embeddings (via openai route/schema) and reranker (via jina route/schema). The cool thing is it uses same model instance for both. Also I managed to easily enable "mps" for any transformer (this is worth separate PR) 
This is now pretty hacky but maybe will manage to provide a PR.

**Reply by mudler** [reply]
That'd be great @ciekawy !
